### ISSUE TITLE
Show a graph of recent changesets with their coverage

### ISSUE THREAD

**Issue by marco-c** [marco-c/code-coverage-reports#28]
Similar to the graph at https://codecov.io/gh/mozilla/grcov.

**Comment by marco-c** [comment]
The full implementation is going to require https://github.com/mozilla/release-services/issues/1226, but you can definitely start in the meantime!

I would create a new page alongside report.html, called graph.html. You can use https://github.com/metricsgraphics/metrics-graphics to create the graph. For now, use placeholder data that you define (date on the X axis, percentage on the Y axis).

**Comment by vipulchhabra99** [comment]
@RasAlgheti i also started working on same issue few days back but couldn't complete it as i have my exams going on which will continue till coming Friday so if you can complete faster you can start working on same

**Comment by vipulchhabra99** [comment]
@macro-c I have few questions regarding the issue
1) would I haved to compress the files to make a pull request?
2) is release-services#1226 issue solved?
3) in first comment you mentioned for report.html but I couldn't found any file of the same name

Thanks

**Comment by marco-c** [comment]
Feel free to both work on the issue, it's an experience anyway even if just one of you will get the PR merged. The description of what needs to be done is in the first few comments.

> would I haved to compress the files to make a pull request?

What do you mean?

> is release-services#1226 issue solved?

No, it is still open.

> in first comment you mentioned for report.html but I couldn't found any file of the same name

It's in the gh-pages branch: https://github.com/marco-c/code-coverage-reports/blob/gh-pages/report.html.

**Comment by vipulchhabra99** [comment]
Oh I actually saw files in master branch which was a compressed file so I thought I would also have to compress to send a PR to it sorry,

**Comment by RasAlgheti** [comment]
Hi, I've been trying to create the page but so far I've had no success getting past an error for which I most likely don't have sufficient expertise to solve (unless it's something glaringly obvious that I'm missing). The errors I keep getting are of this type, varying between attribute x and d:
> <rect> attribute x: Expected length, "NaN".
It looks like somewhere along the process of creating the graph, the data of the JSON object I feed it is lost somehow.

I also thought it'd be worth mentioning that it seems that support for MetricsGraphics.js has been mostly dropped for a while now, with members of the project themselves suggesting to use Chart.js instead of MetricsGraphics (this was 2 months ago in the project's IRC).

**Comment by marco-c** [comment]
> I also thought it'd be worth mentioning that it seems that support for MetricsGraphics.js has been mostly dropped for a while now, with members of the project themselves suggesting to use Chart.js instead of MetricsGraphics (this was 2 months ago in the project's IRC).

Are you sure? I couldn't find such information. @wlach is MetricsGraphics still maintained?

**Comment by RasAlgheti** [comment]
The information I was referring to was in this log of the IRC: [https://mozilla.logbot.info/metricsgraphics/20181015](url)

I'm sorry if it sounded like I was claiming that MetricsGraphics had completely stopped it's maintenance, it was just to mention that the project's activity was lower in recent months and that it was possible that the error was due to something that needed updating in MetricsGraphics. I still think it's extremely likely that the error was in something I did wrong.

**Comment by RasAlgheti** [comment]
As I thought, it was something obvious that I missed. 😅 
Here's a screenshot of the graph using a small css file I made and with some placeholder data:
![graph](https://user-images.githubusercontent.com/22740773/49683333-fec53c00-faba-11e8-9061-b91056d4a9a1.PNG)
I was also wondering if you were looking more for a partial that you can insert into other html files or if you want a full page like https://codecov.io/gh/mozilla/grcov. Also, do you want the time frame selector like in the codecov website as well, or is that another issue altogether?

**Comment by RasAlgheti** [comment]
Hey, sorry for the late answer. I already have the modified report.html and tested locally, the only issue is that this is my first time creating a pull request and I'm not exactly sure how to go about doing it. 😅
So, sorry to bother you but for the sake of me not somehow breaking something, a quick rundown would help me a great deal.

**Comment by marco-c** [comment]
Let me point you to some documentation that you can read:
https://guides.github.com/introduction/flow/
https://help.github.com/articles/about-branches/
https://help.github.com/articles/about-pull-requests/

After reading this, you should have a basic understanding of the workflow, how to submit PRs, and so on.

**Comment by marco-c** [comment]
Still pending.

**Comment by vipulchhabra99** [comment]
ok i am ready to do work .i have one doubt that the metricsgraphics require gulp so i have to include thode files or not?
thanks in advance

**Comment by marco-c** [comment]
Are you sure metricsgraphics requires gulp in order to be used? Maybe it's required to build it, but I'd find it strange if it were required for usage.

**Comment by vipulchhabra99** [comment]
ok i will check once more and where the graph needs to be placed below the reports or above it?

**Comment by marco-c** [comment]
I'd say above.

**Comment by vipulchhabra99** [comment]
@marco-c  should i send the pr?

**Comment by vipulchhabra99** [comment]
@marco-c should i send ??

**Comment by sylvestre** [comment]
Marco is in holiday. Yes, please submit the pr, it looks good :)

**Comment by vipulchhabra99** [comment]
thank you @sylvestre sorry i didn't knew about the same .do you know when he will be back?

**Comment by sylvestre** [comment]
Soon. You can still propose the PR now !

**Comment by vipulchhabra99** [comment]
ok thanks @sylvestre

**Comment by vipulchhabra99** [comment]
@marco-c please review my pr for this issue :smile:

**Comment by vipulchhabra99** [comment]
@marco-c you were right that gulp was used only for its built. :smile:

**Comment by vipulchhabra99** [comment]
@marco-c please review my pr for this issue :smile: so that i can start working on other issue

**Comment by marco-c** [comment]
@vipulchhabra99 I'll review it in a bit, but you can already start working on the other issue, you can work on it in a different branch (see https://guides.github.com/introduction/flow/).

**Comment by vipulchhabra99** [comment]
@marco-c i have a confusion that if create a branch to current head and if any changes are required in previous commit then i think it will create a merge conflict .sorry at that time i forgot to create a different branch but fron onwards i will take care of it

**Comment by vipulchhabra99** [comment]
@marco-c please review my pr

**Comment by marco-c** [comment]
@vipulchhabra99 I will review it soon, just very busy this week :)
There's no rush though, you can start working on the other issue :smiley:

**Comment by vipulchhabra99** [comment]
ya sure @marco-c  but please check my comment  https://github.com/marco-c/code-coverage-reports/issues/26#issuecomment-451414947

**Comment by vipulchhabra99** [comment]
hello @marco-c  please check https://github.com/marco-c/code-coverage-reports/issues/26#issuecomment-452924909 my development part for this issue is complete only testing part is left please provide so that i can check if functions work correctly

**Comment by vipulchhabra99** [comment]
@sylvestre https://github.com/marco-c/code-coverage-reports/issues/26#issuecomment-452924909 do you something regarding this?

**Comment by marco-c** [comment]
@vipulchhabra99 please stop asking questions on this issue related to other issues. It makes the history of the issues really hard to follow.

**Comment by hmittal657** [comment]
Hi, this is my first time trying to contribute to opensource, can someone guide me through the process, 
do I fork it in my profile and then work on it?

**Comment by marco-c** [comment]
@hmittal657 yes. I suggest reading a bit about the GitHub workflow.

**Comment by vipulchhabra99** [comment]
Hey @marco-c please review my pr if any changes are required please do mention so that I can update necessary changes

**Comment by Shivansh2407** [comment]
Hey @marco-c Is the work Still Pending?If Yes Can You Please Explain Me In Brief So That I Can Help You Out .
Thank You For Your Cooperation

**Comment by marco-c** [comment]
@vipulchhabra99 has a PR for this, the only thing missing is fixing the dependency (https://github.com/mozilla/release-services/issues/1226), which will take some time.

**Comment by ashutoshmishraji** [comment]
hi @marco-c I want to contribute in this. please guide me

**Comment by vipulchhabra99** [comment]
@ashutoshmishraji i have already submitted a pr for same so you can try fixing other bugs

**Comment by marco-c** [comment]
Fixed by #36.
